### DISCUSSION TITLE
Issues with Magick++ on MacOS

### DISCUSSION THREAD

**Discussion by tiagoovieira** [ImageMagick/ImageMagick#3587] [Help]
I'm trying to build a simple C++ app in MacOS (Big Sur) using ImageMagick

This is my CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.7)
project(HelloWorld)
add_definitions(-std=c++17)

set(CXX_FLAGS "-Wall")
set(CMAKE_CXX_FLAGS, "${CXX_FLAGS}")

add_executable(HelloWorld src/main.cpp)

add_definitions( -DMAGICKCORE_QUANTUM_DEPTH=16 )
add_definitions( -DMAGICKCORE_HDRI_ENABLE=0 )
find_package(ImageMagick COMPONENTS Magick++)
include_directories(${ImageMagick_INCLUDE_DIRS})
target_link_libraries(HelloWorld ${ImageMagick_LIBRARIES})
```
and on my main.cpp, I just have #include <Magick++.h> at the top of the file

When I run the build I get the
```
/usr/local/Cellar/imagemagick/7.0.10-48/include/ImageMagick-7/Magick++/Functions.h:14:10: fatal error: 'string' file not found
```
any ideas what's the issue here? I reinstalled imagemagick but I still get the same error.

CMake version is 3.19

**Comment by urban-warrior** [comment]
Can you try the latest release of ImageMagick, 7.0.11-8?  Does it return the same exception?  We verified the Magick++ demo compiles without complaint on our Fedora OS system.

**Comment by dlemstra** [comment]
It looks like CMAKE is unable to find this header file: `#include <string>`. No clue why that is happening though. Maybe you are missing some files on your system?
